### PR TITLE
Allow users to set the architecture as a flag

### PR DESCRIPTION
**Pull request by 1byte2bytes**
Hello! I submit this patch for you to review because I am annoyed by the constant prompt to choose if I want M1 or not while attempting to download several different versions back to back. I also want to make the program able to take only arguments and not console input so it can be run by a script that automatically downloads new versions for me.

Please let me know if you have any questions, comments, or changes.

Thanks for your time!

### CODE REVIEW

**Comment by Drovosek01** [comment]
Thank you, but how it works?
I try, but...
Снимок экрана 2021-11-05 в 18 56 58

**Comment by hrvstr** [comment]
> Hello! I submit this patch for you to review because I am annoyed by the constant prompt to choose if I want M1 or not while attempting to download several different versions back to back. I also want to make the program able to take only arguments and not console input so it can be run by a script that automatically downloads new versions for me.

Nice idea, it might make sense to add another argument `--latest`. Otherwise how do you know the latest version if you don't run it interactively?

> Thank you, but how it works?

Not sure if running the `.command` file passes all the arguments to the python file. Did you try to run ccdl.py directly with the arguments like `./ccdl.py -a arm64`

**Comment by 1byte2bytes** [comment]
@Drovosek01 

> Thank you, but how it works?
> I try, but...

Hmmm, that is interesting, it doesn't work when you call the `.command` file. I was passing it directly to the Python file the whole time, like `python3 ccdl.py -a arm64`. I would have to look into why the .command one isn't parsing it properly.

---

@hrvstr 

> Nice idea, it might make sense to add another argument `--latest`. Otherwise how do you know the latest version if you don't run it interactively?

I could add that, what I am doing is archiving all versions of the Adobe software I use as I'm a data hoarder and like to keep the history of software. This would be useful to have though, for me and I'm sure for others.

> Not sure if running the `.command` file passes all the arguments to the python file. Did you try to run ccdl.py directly with the arguments like this `./ccdl.py -a arm64`

Yeah, see my comment to Drovosek01. I didn't test that part, and it doesn't pass through it seems.

File: ccdl.py [hunk 1]
```diff
@@ -332,10 +332,17 @@ def runccdl():
         print('Adobe HyperDrive installer not found.\nPlease make sure the Creative Cloud app is installed.')
         exit(1)
 
-    if platform.machine() == 'arm64':
-        ism1 = questiony('Do you want to make M1 native packages')
+    if args.arch:
+        if args.arch == "arm64" or args.arch == "m1" or args.arch == "as":
```
Comment (thpryrchn) [inline ccdl.py:336]: change `if args.arch ==` to `if args.arch.lower() ==` so Capitalization don't matter.

**Review by thpryrchn — COMMENTED** [review]
Seems Good, except for one minor change.

**Pull request merged**